target/cheri-common/op_helper_cheri_common.c

```diff
@@ -959,23 +999,32 @@ const cap_register_t *get_load_store_base_cap(CPUArchState *env, uint32_t cb)
 #endif
 }
 
-static inline target_ulong cap_check_common(uint32_t required_perms,
-                                            CPUArchState *env, uint32_t cb,
-                                            target_ulong offset, uint32_t size,
-                                            uintptr_t _host_return_address)
+target_ulong cap_check_common_reg(uint32_t required_perms, CPUArchState *env,
+                                  uint32_t cb, target_ulong offset,
+                                  uint32_t size, uintptr_t _host_return_address,
+                                  const cap_register_t *cbp,
+                                  uint32_t alignment_required,
+                                  unaligned_memaccess_handler unaligned_handler)
 {
-    const cap_register_t *cbp = get_load_store_base_cap(env, cb);
+
+#define CHECK_PERM(X) ((required_perms & ~cbp->cr_perms) & (X))
```

Comment:
This is a bit weird, I'd expect it not to negate so callers do !CHECK_PERM

Reply:
I had to stare at this for a while. Possibly easier to just use the old logic and rely on the compiler to optimize it.